### PR TITLE
some refractoring

### PR DESCRIPTION
**Pull request by wanderer**

### CODE REVIEW

**Comment by wanderer** [comment]
@kumavis renames create -> createInstance

File: kernel.js [hunk 1]
```diff
@@ -91,15 +95,30 @@ module.exports = class Kernel {
...
       }
     }
+
+    if (responsePort) {
+      this.send(responsePort, new Message({
+        data: result
+      }))
+      this.ports._unboundPorts.add(responsePort)
+    }
+
```
Comment (kumavis) [inline kernel.js:105]: shouldnt be part of PR?

File: kernel.js [hunk 2]
```diff
@@ -91,15 +95,30 @@ module.exports = class Kernel {
...
     this.ports.clearUnboundedPorts()
-    // const responsePort = this.message.responsePort
-    // if (responsePort) {
-    //   this.send(responsePort, new Message({data: result}))
-    // }
     this._runNextMessage()
     return result
   }
 
+  getResponsePort (message) {
```
Comment (kumavis) [inline kernel.js:111]: hmm?

File: portManager.js [hunk 3]
```diff
@@ -270,4 +246,20 @@ module.exports = class PortManager {
...
+    message.ports.forEach(port => this._unboundPorts.add(port))
+  }
+
+  checkSendingPorts (message) {
+    for (const port of message.ports) {
+      if (this.isBound(port)) {
+        throw new Error('message must not contain bound ports')
+      }
+    }
+  }
```
Comment (kumavis) [inline portManager.js:264]: nice helpers

**Review by kumavis — COMMENTED** [review]
looks good but theres some diffs you said you didnt expect to be in the PR

**Pull request merged**